atoMEC/models.py

```diff
@@ -15,6 +25,50 @@
 
 
 class ISModel:
+    """
+    The ISModel represents a particular family of AA models known as ion-sphere models.
+
+    The implementation in atoMEC is based on the model described in the following pre-print:
+    T. J. Callow, E. Kraisler, S. B. Hansen, and A. Cangi, (2021).
+    First-principles derivation and properties of density-functional average-atom models. arXiv preprint arXiv:2103.09928.
+
+    Parameter inputs for this model are related to particular choices of approximation, e.g. boundary conditions
+    or exchange-correlation functional, rather than fundamental physical properties.
+
+    Parameters
+    ----------
+    atom : :obj:`Atom`
+        The :obj:`Atom` object
+    xfunc : str or int, optional
+        The exchange functional, can be the libxc code or string, or special internal value
+        Default : "lda_x"
+    cfunc : str or int, optional
+        The correlation functional, can be the libxc code or string, or special internal value
+        Default : "lda_c_pw"
+    bc : str, optional
+        The boundary condition, can be "dirichlet" or "neumann"
+        Default : "dirichlet"
+    spinpol : bool, optional
+        Whether to run a spin-polarized calculation
+        Default : False
+    spinmag : int, optional
+        The spin-magentization
+        Default: 0 for nele even, 1 for nele odd
+    unbound : str, optional
+        The way in which the unbound electron density is computed
+        Default : "ideal"
+    write_info : bool, optional
+        Writes information about the model parameters
+        Default : True
+
+    Attributes
+    ----------
+    nele_tot: int
+        total number of electrons
+    nele: array_like
```

Comment:
I see `nele_tot` to be defined on line 85. But where is attribute `nele` defined?

Reply:
It's defined in the `spinpol` and `spinmag` setters (since it needs to change if either of these is updated - I guess there may be a better way to do that).

![Screenshot from 2021-07-02 14-42-13](https://user-images.githubusercontent.com/18050753/124276259-d8b58a80-db43-11eb-8737-95aff550f06f.png)

Reply:
Ah, I've overlooked this. So that is fine. Maybe one can define `nele` also as property with a setter function which than can be reused in the setters of `spinmag` and `spinpol`? This would at least avoid code duplication. Why or when does the try block raise an `AttributeError` and why is this then passed? (do be honest, I am just wondering and asking, it might be all okish :))

Reply:
That sounds like the way to go.

Because `nele` needs both the `self.spinmag` and `self.spinpol` input, `AttributeError` is raised if one has been set but not yet the other. It is passed because `nele` will always be set when the as-yet-unset of these variables is later set. I guess this problem would be avoided with your suggestion.

Reply:
I will merge this PR now, as the discussion is going in a different direction (maybe worth another issue, lets see).